sdks/python/apache_beam/runners/portability/fn_api_runner/translations.py

```diff
@@ -541,6 +557,14 @@ def add_parent(child, parent):
     components.transforms[transform_id].CopyFrom(transform)
     add_parent(transform_id, stage.parent)
 
+  if partial:
```

Comment:
I think we only want to copy sub-transforms of the transforms from line 552 above, perhaps move this logic up there? (Just a thought, maybe having the method be called (`copy_subtransforms` and having it copy (recursively) just the subtransforms could make this even easier to follow).

Reply:
Moved it back up and cleaned up the method.